src/Factory/LinkFactory.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\GraphQL\Search\Factory;
+
+use Atoolo\GraphQL\Search\Resolver\UrlRewriter;
+use Atoolo\GraphQL\Search\Resolver\UrlRewriterType;
+use Atoolo\GraphQL\Search\Types\Link;
+use Atoolo\Resource\Resource;
+
+class LinkFactory
+{
+    public function __construct(
+        private readonly UrlRewriter $urlRewriter,
+    ) {}
+
+    public function create(Resource $resource): Link
+    {
+        return new Link(
+            $this->getUrl($resource),
+            $this->getLabel($resource),
+            $this->getAriaLabel($resource),
+            $this->getDescription($resource),
+            $this->opensNewWindow($resource),
+            $this->isExternal($resource),
+        );
+    }
+
+    protected function getUrl(Resource $resource): string
+    {
+        return $this->isMedia($resource)
+            ? $this->urlRewriter->rewrite(
+                UrlRewriterType::MEDIA,
+                $resource->data->getString('mediaUrl'),
+            )
+            : $this->urlRewriter->rewrite(
+                UrlRewriterType::LINK,
+                $resource->location,
+            );
+    }
+
+    protected function getLabel(Resource $resource): ?string
+    {
+        return $resource->data->getString(
+            'external.label',
+            $resource->data->getString(
+                'base.title',
+                $resource->data->getString('name'),
+            ),
+        );
+    }
+
+    protected function getAriaLabel(Resource $resource): ?string
+    {
+        return $resource->data->has('external.accessibilityLabel')
+            ? $resource->data->getString('external.accessibilityLabel')
+            : null;
+    }
```

Comment:
Why was `arialLabel` chosen as the term here and not `accessibilityLabel`? Couldn't this label also be used in a different context outside the ARAI specification?

Reply:
You're right, I renamed it to `aaccessibilityLabel`.